docs/source/compilers.rst

```diff
@@ -65,10 +65,7 @@ lists correctly: braced initializer lists are not properly matched to static
 arrays. This prevent compile-time detection of the length of a braced
 initializer list.
 
-A consequence is that we need to use stack-allocated shape types in these cases.
-Workarounds for this compiler bug arise in various files of the code base.
-Everywhere, the handling of `Clang < 3.8` is wrapped with checks for the
-``X_OLD_CLANG`` macro.
+Those versions of the Clang compiler are not supported anymore.
```

Comment:
I think you should keep the old paragraph, and add that we drop support for clang < 4.0 in xtensor 0.22

Reply:
Sure, will do that! Thanks

Reply:
Out of curiosity, up to which year does dropping clang < 4.0 mean? More specifically, I am never really confident on which clang version is really being used on OSx, but it seems that clang < 4.0 has not in been use since forever? See https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)

Reply:
@JohanMabille will have more insights about this !

Reply:
The Apple versioning of clang is different from the official / real version of clang. To know wihch version of clang is used, you must actually check the version of LLVM. According to the Toolchain Version History from your link, this means clang < 4 was used up to XCode 9 (excluded).

Also notice that we actually drop support for clang < 3.8 (meaning up to XCode 8.0 excluded). But since the CI only tests starting from clang 4.0, we prefer to be conservative here. However, xtensor should still work with clang 3.8 and clang 3.9.